sql/ddl_api.sql

```diff
@@ -91,11 +91,33 @@ BEGIN
     WHERE attrelid = main_table AND attname = time_column_name;
 
     -- Timestamp types can use default value, integral should be an error if NULL
-    IF time_type IN ('TIMESTAMP', 'TIMESTAMPTZ', 'DATE') AND chunk_time_interval IS NULL THEN
-        chunk_time_interval_actual := _timescaledb_internal.interval_to_usec('1 month');
-    ELSIF time_type IN ('SMALLINT', 'INTEGER', 'BIGINT') AND chunk_time_interval IS NULL THEN
-        RAISE EXCEPTION 'chunk_time_interval needs to be explicitly set for types SMALLINT, INTEGER, and BIGINT'
-        USING ERRCODE = 'IO102';
+    IF time_type IN ('TIMESTAMP', 'TIMESTAMPTZ', 'DATE') THEN
```

Comment:
I think this time_interval stuff should be moved to a properly named function that converts the time_interval from whatever type it is to an internal type. The `create_hypertable()` function is already pretty messy as is. A separate function is also easier to test more extensively.